corehq/apps/app_manager/static/app_manager/js/app_manager.js

```diff
@@ -282,6 +288,27 @@ hqDefine('app_manager/js/app_manager', function () {
             }
             childList.append(childModule);
         }
+        function moveModuleToBottom(module) {
+            $("ul.appnav-module").append(module);
+        }
+        function initChildModuleUpdateListener() {
+            // If a child module is created or removed, update the sidebar
+            $('#module-settings-form').on('saved-app-manager-form', function () {
+                var modulesByUid = getModulesByUid(),
+                    module = modulesByUid[$('#module-settings-form').data('moduleuid')],
+                    oldRoot = $(module).data('rootmoduleuid'),
+                    newRoot = $('select[name=root_module_id]').val();
```

Comment:
Nitpick: do a `.find` inside the form instead of a global search.

Reply:
excellent point - just pushed that change along with a few similar changes.  I was toying with the idea of making a top-level `$scope` var and doing everything with respect to that, but figured that might be getting carried away for now.